sympy/assumptions/ask.py

```diff
@@ -9,66 +9,1114 @@
+
+    @memoize_property(predicate_storage)
+    def integer(self):
+        """
+        Integer predicate.
+
+        ``Q.integer(x)`` is true iff ``x`` belongs to the set of integer numbers.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, S
+        >>> ask(Q.integer(5))
+        True
+        >>> ask(Q.integer(S(1)/2))
+        False
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Integer
+
+        """
+        return Predicate('integer')
+
+    @memoize_property(predicate_storage)
+    def rational(self):
+        """
+        Rational number predicate.
+
+        ``Q.rational(x)`` is true iff ``x`` belongs to the set of
+        rational numbers.
+
+        Examples
+        ========
+
+        >>> from sympy import ask, Q, pi, S
+        >>> ask(Q.rational(0))
+        True
+        >>> ask(Q.rational(S(1)/2))
+        True
+        >>> ask(Q.rational(pi))
+        False
+
+        References
+        ==========
+
+        https://en.wikipedia.org/wiki/Rational_number
+
+        """
+        return Predicate('rational')
+
+    @memoize_property(predicate_storage)
+    def irrational(self):
+        """
+        Irrational number predicate.
+
+        ``Q.irrational(x)`` is true iff ``x``  is any real number that
+        cannot be expressed as a ratio of integers.
+
+        Examples
+        ========
+
+        >>> from sympy import ask, Q, pi, S
+        >>> ask(Q.irrational(0))
+        False
+        >>> ask(Q.irrational(S(1)/2))
+        False
+        >>> ask(Q.irrational(pi))
+        True
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Irrational_number
+
+        """
+        return Predicate('irrational')
+
+    @memoize_property(predicate_storage)
+    def finite(self):
+        """
+        Finite predicate.
+
+        ``Q.finite(x)`` is true if ``x`` is neither an infinity
+        nor a ``NaN``. In other words, ``ask(Q.finite(x))`` is true for all ``x``
+        having a bounded absolute value.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, Symbol, S, oo
+        >>> x = Symbol('x')
+        >>> ask(Q.finite(S.NaN))
+        False
+        >>> ask(Q.finite(oo))
+        False
+        >>> ask(Q.finite(1))
+        True
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Finite
+
+        """
+        return Predicate('finite')
+
+    @memoize_property(predicate_storage)
     @deprecated(useinstead="finite", issue=9425, deprecated_since_version="0.7.7")
     def bounded(self):
+        """
+        See documentation of ``Q.finite``.
+        """
         return Predicate('finite')
 
-    @classproperty
+    @memoize_property(predicate_storage)
+    def infinite(self):
+        """
+        Infinite number predicate.
+
+        ``Q.infinite(x)`` is true iff the absolute value of ``x`` is
+        arbitrarily large.
```

Comment:
That's a shaky definition. If we're talking about absolute value, couldn't we just say that it is infinity? 

Also this needs examples. `oo`, `-oo`, `zoo`, and other "complex infinites" like `oo*I` or `2 + oo*I` are all the ones I know about. They also should show that `nan` is not considered infinite.